include/miopen/miopen.h

```diff
@@ -354,9 +356,14 @@ typedef enum
         4, /*!< Pack of four 8-bit int points in NCHW_VECT_C format (Partially supported) */
     miopenBFloat16 = 5, /*!< 16-bit binary floating point (8-bit exponent, 7-bit fraction)
                            (Partially supported) */
-    miopenDouble  = 6,  /*!< 64-bit floating point (Partially supported) */
+    miopenDouble = 6,   /*!< 64-bit floating point (Partially supported) */
+#ifdef MIOPEN_BETA_API_FP8
     miopenFloat8  = 7,
-    miopenBFloat8 = 8
+    miopenBFloat8 = 8,
+#else
```

Comment:
I think the library won't compile without MIOPEN_BETA_API_FP8 because `miopenFloat8` and `miopenBfloat8` are used in many places.

Reply:
Just so, MIOpen wouldn't compile without MIOPEN_BETA_API_FP8. The compilation variable was added to the CMakeLIsts.txt so that miopen would always build with it. This would however prevent the use of these enum values by a simple inclusion of the header without the defines argument.
But perhaps it would be better form to leave the enumerations without the guards and only keep them around the functions?

Reply:
[resolved]